BrawlLib/Wii/Animations/CHR0SMDImporter.cs

```diff
@@ -0,0 +1,175 @@
+﻿using BrawlLib.SSBB.ResourceNodes;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace BrawlLib.Wii.Animations
+{
+    //This file was wrote by RedStoneMatt, because he needed that importer for a little mod.
+    //It ended up not even working.
+    //Yikes.
+    internal class CHR0SMDImporter
+    {
+        private class SMDFrame
+        {
+            public int id, boneID;
+            public float posx, posy, posz, rotx, roty, rotz;
+
+            public SMDFrame()
+            {
+                id = new int();
+                boneID = new int();
+                posx = new float();
+                posy = new float();
+                posz = new float();
+                rotx = new float();
+                roty = new float();
+                rotz = new float();
+            }
+
+            public SMDFrame(int ID, int BoneID, float XPos, float YPos, float Zpos, float XRot, float YRot, float ZRot)
+            {
+                id = ID;
+                boneID = BoneID;
+                posx = XPos;
+                posy = YPos;
+                posz = Zpos;
+                rotx = XRot;
+                roty = YRot;
+                rotz = ZRot;
+            }
+        }
+
+        public static CHR0Node Convert(string input)
+        {
+            try
+            {
+                CHR0Node chr0 = new CHR0Node
+                {
+                    Name = Path.GetFileNameWithoutExtension(input),
+                    Loop = false,
+                    Version = 4
+                };
+                List<string> boneNames = new List<string>();
+                List<int> boneIDs = new List<int>();
+                int lineNumber = 0;
+                bool isReadingBones = false;
+                bool isReadingFrames = false;
+                int currentFrameID = new int();
+                float radianFloat = 57.29579143313326f; //Angles are given in radians. To convert them into degrees, multiply them by this float.
+                List<SMDFrame> allFrames = new List<SMDFrame>();
+                for (TextReader reader = new StreamReader(input); reader.Peek() != -1;) //Writing data into Lists
```

Comment:
lineNumber should be incremented in the for loop declaration, or this should be changed to a while loop